### PR TITLE
refactor: use umi 3

### PR DESCRIPTION
**Pull request by ycjcl868**
close https://github.com/ant-design/ant-design-pro/issues/6041
close https://github.com/ant-design/ant-design-pro/issues/6060

### CODE REVIEW

**Comment by kaoding** [comment]
弹动一下

**Comment by afc163** [comment]
Ant Design Pro 要升级到 5.0，老的 4.0 切分支。

**Comment by tomgao365** [comment]
看来，跳过 3.0 有的后遗症，无法和 antd 版本相呼应了。

**Comment by xiaohuoni** [comment]
我就说为什么没有3.0（手动狗头）

**Comment by zhou-ting** [comment]
> Ant Design Pro 要升级到 5.0，老的 4.0 切分支。

请问大概什么时候能出5.0能？非常期待

**Comment by kaoding** [comment]
> 看来，跳过 3.0 有的后遗症，无法和 antd 版本相呼应了。

把5.0改叫3.0，弥补没有3.0的遗憾。

**Comment by chenshuai2144** [comment]
- [ ]  umi-plugin-antd-icon-config
- [ ]   umi-plugin-antd-theme
- [ ]   umi-plugin-pro-block
- [ ]   umi-plugin-react
- [ ]   umi-request
- [ ]   umi-plugin-ga

**Comment by ycjcl868** [comment]
`@umijs/preset-react` 包含： umi-plugin-react 和 umi-plugin-ga

**Comment by bubucuo** [comment]
按照官网说的升级之后，还是报了这个错，这个应该是框架里的不兼容，请问有修改的方法么，在线急等。。。

Error: Reducer "login" returned undefined during initialization. If the state passed to the reducer is undefined, you must explicitly return the initial state. The initial state may not be undefined. If you don't want to set a value for this reducer, you can use null instead of undefined.

**Comment by ycjcl868** [comment]
- umi-plugin-antd-icon-config 
- umi-plugin-antd-theme
- umi-plugin-pro-block

这几个插件没升到 umi 3，没有 options 参数了 @chenshuai2144

File: package.json [hunk 1]
```diff
@@ -104,7 +105,9 @@
     "prettier": "^1.19.1",
     "pro-download": "1.0.1",
     "serverless-http": "^2.0.2",
-    "stylelint": "^13.0.0"
+    "stylelint": "^13.0.0",
+    "umi-plugin-antd-icon-config": "^2.0.0-1",
+    "umi-plugin-antd-theme": "^2.1.0"
```
Comment (afc163) [inline package.json]: 这些插件能收到 presets 里么。

Comment (ycjcl868) [reply package.json]: https://github.com/umijs/plugins/pull/81/files，还没发，发了我这边改

File: config/config.ts [hunk 2]
```diff
@@ -12,10 +12,24 @@ const { winPath } = utils;
 const { ANT_DESIGN_PRO_ONLY_DO_NOT_USE_IN_YOUR_PRODUCTION, REACT_APP_ENV } = process.env;
 const isAntDesignProPreview = ANT_DESIGN_PRO_ONLY_DO_NOT_USE_IN_YOUR_PRODUCTION === 'site';
 
-// TODO:
-// umi-plugin-pro-block umi-plugin-antd-theme umi-plugin-antd-icon-config 需要加
-// @umijs/plugin-analytics 日志 log 应该开启后打印
-// **/_mock.js 需不需要支持？
+// if (isAntDesignProPreview) {
```
Comment (chenshuai2144) [inline config/config.ts]: 直接删掉吧

**Comment by kaoding** [comment]
升 umi 3 碰到了什么比较难的问题吗？这个PR看着不怎么动了。

File: mock/listTableList.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // eslint-disable-next-line import/no-extraneous-dependencies
 import { Request, Response } from 'express';
 import { parse } from 'url';
-import { TableListItem, TableListParams } from './data.d';
+import { TableListItem, TableListParams } from '../src/pages/ListTableList/data.d';
```
Comment (afc163) [inline mock/listTableList.ts]: 不需要 `.d` 结尾吧？

Comment (chenshuai2144) [reply mock/listTableList.ts]: 可以的，是我的个人习惯

Comment (afc163) [reply mock/listTableList.ts]: .d.ts 如果没开自动编译的话可能不存在的。

Comment (chenshuai2144) [reply mock/listTableList.ts]: 这个是我自己写的 d.ts
是不是除了 typing.d.ts 不允许自己写 d.ts

**Comment by chenshuai2144** [comment]
@ycjcl868  看一下 ci 问题，windows 过不去

**Comment by qiqizjl** [comment]
多久能合并呢？

**Comment by ycjcl868** [comment]
merge 下 master，windows ci 看起来需要判断下 server

image

**Comment by ycjcl868** [comment]
重新触发下 ci

**Comment by joesdu** [comment]
合并上去了吗？

**Comment by joesdu** [comment]
感谢大佬对Umi3的升级支持.

**Comment by kaoding** [comment]
感谢大佬对Umi3的升级支持.

**Comment by chenshuai2144** [comment]
@kaoding  @joesdu  写好迁移文档就合并。

**Comment by tomgao365** [comment]
看来下周就可以在新项目用起来了。

**Comment by mdesmet** [comment]
Tests still not work.

I changed umi test to umi-test but there seems to be no default configuration anymore. (in umi 2 the jest setup was merged with  extraSetupFiles: ['./tests/setupTests.js'],).

**Comment by mdesmet** [comment]
https://github.com/mdesmet/ant-design-pro/tree/refactor-umi-3

I've added a test in src/components.

However failing:

FAIL  src/components/SelectLang/__tests__/index.test.js
  ● Test suite failed to run

    Cannot find module '@/components/SelectLang' from 'index.test.js'

      2 | import { render } from 'enzyme';
      3 | import * as renderer from 'react-test-renderer';
    > 4 | import SelectLang from '@/components/SelectLang';
        | ^
      5 | 
      6 | describe('SelectLang', () => {
      7 |   it('renders correctly', () => {

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:276:11)
      at Object.<anonymous> (src/components/SelectLang/__tests__/index.test.js:4:1)


I managed to get farther by setting  "esModuleInterop": true, in tsconfig.json but then it fails on the JSX template:

  Cannot find module '@/components/SelectLang' from 'index.test.js'

      2 | import { render } from 'enzyme';
      3 | import * as renderer from 'react-test-renderer';
    > 4 | import SelectLang from '@/components/SelectLang';
        | ^
      5 | 
      6 | describe('SelectLang', () => {
      7 |   it('renders correctly', () => {

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:276:11)
      at Object.<anonymous> (src/components/SelectLang/__tests__/index.test.js:4:1)

**Comment by joesdu** [comment]
😢看起来仿佛又过不去了

**Comment by chenshuai2144** [comment]
@ycjcl868  
image

**Comment by afc163** [comment]
umi2 的版本要切分支出来。

umi3 版本是否要升级 pacakge.json 里的版本。

**Comment by 426-330** [comment]
image

**Pull request merged**